app/views/devise/sessions/new.html.erb

```diff
@@ -32,6 +32,9 @@
         required: true,
         wrapper_html: { class: 'margin-top-6' },
       ) %>
+  <% if AbTests::KEY_PAIR_GENERATION.bucket(SecureRandom.uuid) == :key_pair_group %>
+    <%= render KeyPairGeneratorComponent.new(location: 'sign_in') %>
+  <% end %>
```

Comment:
No strong preference from me, but we could consider using ViewComponent's built-in conditional rendering support for this?

https://viewcomponent.org/guide/conditional_rendering.html

Reply:
+1

Reply:
Nice! Done in e9317e724